diff --git a/src/cmake/externalpackages.cmake b/src/cmake/externalpackages.cmake
index 2b5ff46..9a96e28 100644
--- a/src/cmake/externalpackages.cmake
+++ b/src/cmake/externalpackages.cmake
@@ -149,6 +149,11 @@ if (NOT OPENCOLORIO_INCLUDES)
     get_target_property(OPENCOLORIO_INCLUDES OpenColorIO::OpenColorIO INTERFACE_INCLUDE_DIRECTORIES)
 endif ()
 include_directories(BEFORE ${OPENCOLORIO_INCLUDES})
+if (USE_OPENCOLORIO AND TARGET Imath::Imath AND OIIO_USING_IMATH STREQUAL "2")
+    message(FATAL_ERROR
+        "OpenColorIO and OpenEXR use incompatible versions of Imath. "
+        "You cannot use openimageio[opencolorio] for this configuration.")
+endif ()
 
 checked_find_package (OpenCV CONFIG
                       DEFINITIONS USE_OPENCV=1)
